apps/backend/src/datasources/stfc/StfcFapDataSource.ts

```diff
@@ -0,0 +1,67 @@
+import { GraphQLError } from 'graphql';
+
+import { Fap } from '../../models/Fap';
+import { UserRoleShortCodeMap } from '../../models/User';
+import {
+  AssignChairOrSecretaryToFapInput,
+  AssignReviewersToFapArgs,
+} from '../../resolvers/mutations/AssignMembersToFapMutation';
+import { FapDataSource } from '../FapDataSource';
+import PostgresFapDataSource from '../postgres/FapDataSource';
+import { StfcUserDataSource } from './StfcUserDataSource';
+
+const stfcUserDataSource = new StfcUserDataSource();
+
+export default class StfcFapDataSource
+  extends PostgresFapDataSource
+  implements FapDataSource
+{
+  async assignChairOrSecretaryToFap(
+    args: AssignChairOrSecretaryToFapInput
+  ): Promise<Fap> {
+    const roles = await stfcUserDataSource.getUserRoles(args.userId);
+
+    if (
+      roles.find((role) => {
+        return (
+          role.shortCode === UserRoleShortCodeMap[args.roleId] ||
+          role.shortCode === 'user_officer'
+        );
+      })
+    ) {
+      return super.assignChairOrSecretaryToFap(args);
+    }
+
+    throw new GraphQLError(
+      `User does not have the correct role ${args.userId}`
+    );
+  }
+
+  async assignReviewersToFap(args: AssignReviewersToFapArgs): Promise<Fap> {
+    const usersWithRoles = await stfcUserDataSource.getUsersRoles(
+      args.memberIds
+    );
+
+    const members = usersWithRoles
```

Comment:
I think something like `ineligibleUserIds` would be a better name for this variable. Members sounds like they're part of a group, but they're the ones we don't want to add because they're not in a group/role.